### ISSUE TITLE
Formatting of Exceptions

### ISSUE THREAD

**Issue by bruce-usab** [w3c/wcag#895]
**Brief description** of issue:
Exceptions in _three_ SC new to 2.1 don’t follow the pattern used by exceptions in SC of 2.0.  It would be better if the exception phrasing was consistent.

> A foolish consistency is the hobgoblin of little minds, adored by little statesmen and philosophers and divines.  (Ralph Waldo Emerson)

**Background**:  WCAG 2.0 SC with exceptions follow one of two patterns.

1) Old 2.0 **SC with a single (in-line) exception** are:

- 1.2.1 Audio-only and Video-only, 1.2.2 Captions (Prerecorded), 1.2.3 Audio Description or Media Alternative (Prerecorded) — All three end with “except when the audio or video is a media alternative…” (Should this be “except _where_”?)
- 1.4.4 Resize Text — Starts with “Except for captions and images of text…” (Should this SC end with the exception?)
- 1.4.7 Low or No Background Audio — Third option ends “with the exception of occasional sounds…” (Should this be edited to use “except where” phrasing?)
- 2.1.1 Keyboard — Ends with “except where the underlying function requires input that…”
- 2.2.3 No Timing — Ends with “except for non-interactive…”
- 2.3.4 Interruptions — Ends with “except interruptions involving an emergency.” (Should this use “except _for_” or “except _where_” phrasing?)
- 2.4.4 Link Purpose (In Context) — Ends with “except where the purpose of the link…”
- 2.4.5 Multiple Ways — Ends with “except where the Web Page is…”
- 2.4.9 Link Purpose (Link Only) — Ends with “except where the purpose of the link…”
- 3.1.2 Language of Parts — Ends with “except for proper names,…” (and three more items).
- 4.1.1 Parsing — Ends with “except where the specifications allow…”

2) Old 2.0 **SC with multiple exceptions** end with “except for the following” and then have a bulleted list.  These are:

- 1.1.1 Non-text Contrast (Uses “except for situations listed below”.  — Should this be edited to use “except for the following”?) 
- 1.2.1 Audio-only and Video-only (Prerecorded) — Note that 1.2.1 is also in list above
- 1.4.3 Contrast (Minimum)
- 1.4.5 Images of Text
- 1.4.6 Contrast (Enhanced)

Note:  There are four other 2.0 SC with lists (some numbered, and some just bullets, it’s not entirely consistent).  Those follow the pattern:   [all | at least one] of the following is true.  These SC are 1.4.7 Low or No Background Audio, 1.4.8 Visual Presentation (breaks the literal pattern, but the idea is the same and could/should be edited), and 2.2.1 Timing Adjustable, 2.2.2 Pause, Stop, Hide.

The only place WCAG 2.0 use the style where a paragraph starts with the word “Exception:” is in the [definition for general flash and red flash thresholds](http://www.w3.org/TR/2008/REC-WCAG20-20081211/Overview.html#general-thresholddef) — “Flashing that is a fine, balanced, pattern…”

3) With 2.1, there are two **new SC that follow the patterns** above:

- 1.4.11 Non-text Contrast
    - First bulleted option ends with “except for inactive components or where…”
    - Second bulleted option ends with “except when a particular presentation…” (Should this be “except _where_”?)
- 2.5.6 Concurrent Input Mechanisms — Ends with “except where the restriction is…” and three options follow. 

4) With 2.1, there are three **new SC that have new and different patterns** for their exceptions.  These are:

- 1.4.10 Reflow — Ends with “Except for parts of the content which…” and it’s is a straightforward edit to move this bit into the main body text of the SC.   (Also note that this SC uses just “for:” [and not “for all of the following:”].)
- 1.4.12 Text Spacing — Ends with “Exception: Human languages and scripts that…” but will take some effort to match the 2.0 pattern.  (Also note that this SC includes a list that could/should be edited to match the “all of the following is true” pattern.)
- 1.4.13 Content on Hover or Focus — Ends with “Exception: The visual presentation of the additional content is…” and will take some effort to match the 2.0 pattern.  (Also note that this SC uses “the following are true” instead of the “_all of_ the following are true”, so that could/should be edited.)

**Proposal**:

Edit SC in fourth group (_i.e._, 1.4.10, 1.4.12, 1.4.13) for consistency with how exceptions are treated in WCAG 2.0 SC.

Corollary issue:  Should other inconsistencies with exception phrasing be addressed as well?  (E.g., use _where_ and **not** _when_.)

Alternative proposal:  Edit all exception language to match the style used by the Revised 508 Standards, _i.e._, the exceptions would be listed after the base requirement.

**Comment by bruce-usab** [comment]
FWIW, I would be happy to format these as one or more pull requests.  But I have been waiting for go / no-go indication...

**Comment by alastc** [comment]
Hi Bruce, it's on the survey for next week to try and get a go/no-go indication. If you could do an example of one of the SCs, that might help the discussion?

**Comment by bruce-usab** [comment]
Will do.

**Comment by bruce-usab** [comment]
*Current:*
**Success Criterion 1.4.12 Text Spacing** [§](https://www.w3.org/TR/WCAG21/#text-spacing)  (Level AA)
In content implemented using markup languages that support the following text style properties, no loss of content or functionality occurs by setting all of the following and by changing no other style property:
* Line height (line spacing) to at least 1.5 times the font size;
* Spacing following paragraphs to at least 2 times the font size;
*Letter spacing (tracking) to at least 0.12 times the font size;
* Word spacing to at least 0.16 times the font size.

Exception:&nbsp; Human languages and scripts that do not make use of one or more of these text style properties in written text can conform using only the properties that exist for that combination of language and script

**Comment by bruce-usab** [comment]
*Proposed:*
**Success Criterion 1.4.12 Text Spacing** §  (Level AA)
For human languages and scripts that make use of one or more of the following text style properties in written text, content implemented using markup languages that support the following text style properties, no loss of content or functionality occurs by setting all of the following and by changing no other style property:
* Line height (line spacing) to at least 1.5 times the font size;
* Spacing following paragraphs to at least 2 times the font size;
* Letter spacing (tracking) to at least 0.12 times the font size;
* Word spacing to at least 0.16 times the font size.

**Comment by bruce-usab** [comment]
*Current:*
**Success Criterion 1.4.13 Content on Hover or Focus** [§](http://www.w3.org/TR/WCAG21/#content-on-hover-or-focus) (Level AA)
Where receiving and then removing pointer hover or keyboard focus triggers additional content to become visible and then hidden, the following are true:
* Dismissable:&nbsp; A mechanism is available to dismiss the additional content without moving pointer hover or keyboard focus, unless the additional content communicates an input error or does not obscure or replace other content;
* Hoverable:&nbsp; If pointer hover can trigger the additional content, then the pointer can be moved over the additional content without the additional content disappearing;
* Persistent:&nbsp; The additional content remains visible until the hover or focus trigger is removed, the user dismisses it, or its information is no longer valid.

Exception:&nbsp; The visual presentation of the additional content is controlled by the user agent and is not modified by the author.

**Comment by bruce-usab** [comment]
*Proposed:*
**Success Criterion 1.4.13 Content on Hover or Focus** [§](http://www.w3.org/TR/WCAG21/#content-on-hover-or-focus) (Level AA)
Except when the visual presentation of the additional content is controlled by the user agent and is not modified by the author, where receiving and then removing pointer hover or keyboard focus triggers additional content to become visible and then hidden, all of the following are true:
* Dismissable:&nbsp; A mechanism is available to dismiss the additional content without moving pointer hover or keyboard focus, unless the additional content communicates an input error or does not obscure or replace other content;
* Hoverable:&nbsp; If pointer hover can trigger the additional content, then the pointer can be moved over the additional content without the additional content disappearing;
* Persistent:&nbsp; The additional content remains visible until the hover or focus trigger is removed, the user dismisses it, or its information is no longer valid.

**Comment by bruce-usab** [comment]
*Current:*
**Success Criterion 1.4.10 Reflow** [§](https://www.w3.org/TR/WCAG21/#reflow) (Level AA)
Content can be presented without loss of information or functionality, and without requiring scrolling in two dimensions for:

* Vertical scrolling content at a width equivalent to 320 CSS pixels;
* Horizontal scrolling content at a height equivalent to 256 CSS pixels.

Except for parts of the content which require two-dimensional layout for usage or meaning.

**Comment by bruce-usab** [comment]
*Proposed:*
**Success Criterion 1.4.10 Reflow** §(https://www.w3.org/TR/WCAG21/#reflow) (Level AA)
Except for parts of the content which require two-dimensional layout for usage or meaning, content can be presented without loss of information or functionality, and without requiring scrolling in two dimensions for:
* Vertical scrolling content at a width equivalent to 320 CSS pixels;
* Horizontal scrolling content at a height equivalent to 256 CSS pixels.

**Comment by bruce-usab** [comment]
Closing, per email between backlog calls.
